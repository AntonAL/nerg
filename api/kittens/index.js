import axios from 'axios';

function getKittens () {
  return axios.get('http://localhost:3001/');
}

export {
  getKittens
};