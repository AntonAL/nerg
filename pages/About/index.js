import React, { Component } from 'react';
import Link from 'next/link';
class About extends Component {
  render() {
    return (
      <div>
        <Link href="/">Go to home</Link>
        <p>about us</p>
      </div>
    )
  }
}
export default About;
