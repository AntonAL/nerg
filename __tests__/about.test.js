import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import About from "../pages/about";
import { getKittens } from "../api/kittens";

describe("With Snapshot Testing", () => {
  it("About shows a loading", () => {
    const component = renderer.create(<About />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Kittens", () => {
  it("Should fetch an array of kittens", async () => {
    const kittens = await getKittens();
    expect(kittens.data).toBeInstanceOf(Array);
  });
});
