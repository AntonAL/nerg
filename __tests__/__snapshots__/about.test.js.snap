exports[`With Snapshot Testing About shows a loading 1`] = `
<div>
  <header>
    <nav>
      <a
        href="/"
        onClick={[Function]}>
         Home 
      </a>
       |
      <a
        href="/about"
        onClick={[Function]}>
         About 
      </a>
    </nav>
  </header>
  <p>
    Loading
  </p>
  <footer>
    This is the footer
  </footer>
</div>
`;
